common/reconciliation/invariants/concreteExecutionExists.go

```diff
@@ -89,11 +95,20 @@ func (c *concreteExecutionExists) Check(execution interface{}) common.CheckResul
 }
 
 func (c *concreteExecutionExists) Fix(execution interface{}) common.FixResult {
-	fixResult, checkResult := checkBeforeFix(c, execution)
+	currentExecution, _ := execution.(*common.CurrentExecution)
+	var runIDCheckResult *common.CheckResult
+	currentExecution, runIDCheckResult = c.validateCurrentRunID(currentExecution)
```

Comment:
Is this needed? The deletion will be a no-op when `runIDCheckResult != nil` I think?

Reply:
Yes. For scanner, the current run ID is set in the execution. For resender, it passes in the domain and workflow ID. This is to set the current run ID in this case.

Reply:
Updated the code after offline sync